include/mouse.h

```diff
@@ -79,4 +119,107 @@ uint8_t MOUSEBIOS_GetSampleRate();
 void MOUSEDOS_BeforeNewVideoMode();
 void MOUSEDOS_AfterNewVideoMode(const bool setmode);
 
+// ***************************************************************************
+// MOUSECTL.COM / GUI configurator interface
+// ***************************************************************************
+
+class MouseInterface;
+class MousePhysical;
+
+class MouseInterfaceInfoEntry final {
+public:
+
+    bool IsEmulated() const;
+    bool IsMapped() const;
+    bool IsMapped(const uint8_t device_idx) const;
+    bool IsMappedDeviceDisconnected() const;
+
+    MouseInterfaceId GetInterfaceId() const;
+    MouseMapStatus GetMapStatus() const;
+    const std::string &GetMappedDeviceName() const;
+    int8_t GetSensitivityX() const; // -99 to +99
+    int8_t GetSensitivityY() const; // -99 to +99
+    uint16_t GetMinRate() const; // 10-500, 0 for none
+    uint16_t GetRate() const; // current rate, 10-500, 0 for N/A
+
+private:
+
+    friend class MouseInterface;
+    MouseInterfaceInfoEntry(const MouseInterfaceId interface_id);
+
+    const uint8_t idx;
+    const MouseInterface &Interface() const;
+    const MousePhysical  &MappedPhysical() const;
+};
+
+class MousePhysicalInfoEntry final {
+public:
+
+    bool IsMapped() const;
+    bool IsDeviceDisconnected() const;
+    const std::string &GetDeviceName() const;
+
+private:
+
+    friend class ManyMouseGlue;
+    MousePhysicalInfoEntry(const uint8_t idx);
+
+    const uint8_t idx;
+    const MousePhysical &Physical() const;
+};
+
+class MouseConfigAPI final {
+public:
+
+    // Always destroy the object once it is not needed anymore
+    // (configuration tool finishes it's job) and we are returning
+    // to normal code execution!
+
+    MouseConfigAPI();
```

Comment:
This would make sense for the current usage (MOUSECTL.COM command), but the class is intended to be used also by the yet-to-be-written GUI. And here it is perfectly possible that user will want to do a different change to DOS mouse driver, and different for PS/2 mouse emulation - thus, at the moment of creating the object, we will have no knowledge regarding what content the list should have.

I propose to leave this unchanged.

Reply:
Sounds good, and thanks for the explanation. I had (incorrectly) thought that the `list_ids` was _all_ of the interfaces available.

> here it is perfectly possible that user will want to do a different change to DOS mouse driver, and different for PS/2 mouse emulation

I guess `list_ids` differs based on driver - so the DOS mouse driver generates a different set of `list_ids` versus the PS/2 emulation?